app/components/dynamic_back_link/view.rb

```diff
@@ -15,32 +14,7 @@ def link_text
     end
 
     def path
-      return view_trainee(trainee) unless origin_pages.any?
-
-      # If you're currently on a 'confirm' page, then you're on an origin page
-      # that has a back button to the _previous_ origin page.
-      if on_origin_page?
-        # Therefore, return the origin page before this one if it's been stored.
-        return origin_pages.length < 2 ? view_trainee(trainee) : rails_path(origin_pages[-2])
-      end
-
-      rails_path(origin_pages.last)
-    end
-
-  private
-
-    attr_reader :trainee, :text
-
-    def rails_path(route)
-      public_send("#{route}_path", trainee)
-    end
-
-    def origin_pages
-      @origin_pages ||= origin_pages_for(trainee)
-    end
-
-    def on_origin_page?
-      current_page == origin_pages.last
+      OriginPage.new(@trainee, session, request).path
```

Comment:
We're using a mix of instance variables and `attr_reader` methods in this component for `trainee`.

Reply:
We a move them to `attr_reader`